sdk/include/opentelemetry/sdk/metrics/min_max_sum_count_aggregator.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "opentelemetry/version.h"
+#include "opentelemetry/metrics/instrument.h"
+#include "opentelemetry/sdk/metrics/aggregator/aggregator.h"
+
+#include <variant>
+#include <vector>
+#include <mutex>
+#include <memory>
+
+namespace metrics_api = opentelemetry::metrics;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+/**
+ * This aggregator stores and maintains a vector of
+ * type T where the contents in the vector are made
+ * up of the minimum value recorded to this instrument,
+ * the maximum value, the sum of all values, and the
+ * count of all values.
+ *
+ * @tparam T the type of values stored in this aggregator.
+ */
+template<class T>
+class MinMaxSumCountAggregator : public Aggregator<T>
+{
+public:
+  explicit MinMaxSumCountAggregator(metrics_api::BoundInstrumentKind kind)
+  {
+    static_assert(std::is_arithmetic<T>::value, "Not an arithmetic type");
+    this->kind_ = kind;
+    this->values_ = std::vector<T>(4, 0); // {min, max, sum count}
+    this->checkpoint_ = this->values_;
+  }
+
+  /**
+   * Receives a captured value from the instrument and applies it to the current aggregator value.
+   *
+   * @param val, the raw value used in aggregation
+   */
+  void update(T val) override
+  {
+    this->mu_.lock();
+
+    if (this->values_[3] == 0 || val < this->values_[0]) // set min
```

Comment:
I personally would prefer to make these indices named constants, and remove the comments.

For example: `this->values_[kMinValueIndex] = value;`

Up to you :)

Reply:
I added this in but left some of the comments as well to reduce confusion as much as possible.